### PR TITLE
fix(tomorrow): honor goal display precision via Beeminder's dueby

### PR DESCRIPTION
**Pull request by narthur**
## Summary
- `buzz tomorrow` was printing values like `steps +10788.140000000001 in 1 day` because `bareminByEndOfTomorrowAt` did its own float addition (today's baremin + per-day rate) and formatted with `%g`.
- Beeminder's API already publishes per-day deltas rounded to each goal's Display Precision under `dueby[<daystamp>].formatted_delta_for_beedroid`. We now use that for due-today goals in the tomorrow view.
- The per-goal "precision" (a.k.a. internal `quantum`) setting isn't exposed as a top-level goal attribute, but `dueby` is — so we honor precision indirectly by trusting Beeminder's pre-rounded strings.
- Falls back to the existing slope-based bump when `dueby` is missing, has <2 entries, or tomorrow's entry has no formatted delta.

Before / after for the reporter's `steps` goal:

```
- steps        +10788.140000000001 in 1 day  1d   tomorrow 11:59 PM
+ steps        +10789 in 1 day               1d   tomorrow 11:59 PM
```

(Bonus accuracy fix: our local bump used `g.Rate` end-of-graph slope, so it produced `10788.14`; Beeminder's own day-by-day computation gives `10788.12` → `+10789`.)

## Test plan
- [x] `go test -run TestBareminByEndOfTomorrow ./...` — 18 subtests pass (14 pre-existing + 4 new: real steps-style precision case, timey goal, single-entry-dueby fallback, empty-FormattedDelta fallback)
- [x] `go vet ./...` clean, `gofmt -l .` clean
- [x] Live run against a real account: every "due today, shown in tomorrow view" line now matches `dueby.formatted_delta_for_beedroid` exactly (steps `+10789`, active `+94`, nasal `+2`, etc.)
- [ ] Reviewer: spot-check `buzz tomorrow` on a goal with a non-integer rate

Two unrelated test failures (`TestExtractTimeSlots`, `TestExtractTimeSlotsAcrossDates`) also fail on `main` — pre-existing, not from this change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for per-day “dueby” entries from Beeminder’s API so the app can surface pre-formatted daily deltas/totals.

* **Bug Fixes**
  * Improved baremin bumping to prefer Beeminder’s pre-formatted tomorrow value when present, with robust fallback to existing calculations.

* **Tests**
  * Added tests covering tomorrow daystamp selection and baremin bumping behavior across edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/PinePeakDigital/buzz/pull/241)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -577,6 +580,31 @@ func bareminByEndOfTomorrowAt(g Goal, now time.Time) string {
...
+	keys := make([]string, 0, len(g.Dueby))
+	for k := range g.Dueby {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	tomorrow := g.Dueby[keys[1]].FormattedDelta
+	if tomorrow == "" {
+		return "", false
+	}
+	return tomorrow, true
```
[the remark this answers is not in the document]
Comment (narthur) [reply main.go]: Good catch — fixed in 8695794. `bareminFromDueby` now computes tomorrow's daystamp directly from `now` and the goal's `deadline` shift (rather than assuming `keys[1]` is tomorrow), and looks it up by key. Added a `TestTomorrowDaystampFor` table covering midnight, +3h, and -3h cutoffs, plus a "dueby keys are only in the past" fallback case in `TestBareminByEndOfTomorrowAt`.

File: main_test.go [hunk 2]
```diff
@@ -395,6 +395,73 @@ func TestBareminByEndOfTomorrowAt(t *testing.T) {
...
+				Rate:     f(8505.140000000001),
+				Runits:   "d",
+				Dueby: map[string]DuebyEntry{
+					"20250115": {FormattedDelta: "+2283"},
+					"20250116": {FormattedDelta: "+10789"},
+					"20250117": {FormattedDelta: "+10789"},
+				},
+			},
+			expected: "+10789 in 1 day",
+		},
```
[the remark this answers is not in the document]
Comment (narthur) [reply main_test.go:416]: Added in b058b7b — new `due today with past+today+tomorrow dueby still picks tomorrow` case explicitly mixes a past daystamp with today and tomorrow and asserts we pick tomorrow's delta. The prior commit (8695794) already covered the all-past fallback case.

File: main.go [hunk 3]
```diff
@@ -577,6 +580,31 @@ func bareminByEndOfTomorrowAt(g Goal, now time.Time) string {
...
+}
```
[the remark this answers is not in the document]
Comment (narthur) [reply main.go]: Already addressed in 8695794 (commented at the same time as your review). The new `bareminFromDueby` takes `now` and looks up the explicit tomorrow daystamp via the helper `tomorrowDaystampFor`, which also honours the goal's `deadline` shift (your suggested diff would miss the daystamp for goals with positive/negative deadlines, e.g. 3am or 9pm cutoffs). Test coverage for that is in `TestTomorrowDaystampFor` plus the "past+today+tomorrow dueby still picks tomorrow" case in `TestBareminByEndOfTomorrowAt` (b058b7b).
